configure.ac

```diff
@@ -151,6 +151,19 @@ AC_SUBST([git_commit_hash], "$GIT_COMMIT_HASH")
 AC_SUBST([DEV_VERSION_SUFFIX])
 AC_SUBST([dist_version], "$DIST_VERSION")
 
+AC_ARG_WITH([mono-framework],
+  AC_HELP_STRING([--with-mono-framework="MONO_FRAMEWORK"],
+  [Set where to look for the mono framework (macOS) [default=/Library/Frameworks/Mono.framework/Versions/Current]]),
+  with_mono_framework=$enableval,
+  with_mono_framework=/Library/Frameworks/Mono.framework/Versions/Current
```

Comment:
Put AC_ARG_WITH in a Darwin conditional block and that way it is Mac specific and does not confuse anyone else.